xml_converter/src/argument_parser.cpp

```diff
@@ -74,6 +76,22 @@ ParsedArguments parse_arguments(int argc, char* argv[]) {
         else if (string(argv[i]) == "--allow-duplicates") {
             parsed_arguments.allow_duplicates = true;
         }
+        else if (string(argv[i]) == "--split-by-category") {
+            if (type != BehaviorType::EXPORT) {
+                cerr << "Error: --split-by-category needs to follow an output argument" << endl;
+                return {};
+            }
+            split_by_category_depth.set_value(0);
+            while (i + 1 < argc && string(argv[i + 1]).find("--") != 0) {
+                if (is_string_valid_integer(argv[i + 1])) {
+                    split_by_category_depth.set_value(std::stoi(argv[++i]));
+                }
+                else {
+                    cerr << "Error: expected an integer after --split-by-category but received " << argv[i + 1] << endl;
+                    return {};
+                }
+            }
```

Comment:
What happens if we have two arguments after `--split-by-category` instead of just 0 or 1.

Add a new unit test for this as well.